client/network/src/config.rs

```diff
@@ -97,54 +95,42 @@ pub struct Params<B: BlockT, H: ExHashT> {
 	pub metrics_registry: Option<Registry>,
 }
 
-bitflags! {
-	/// Bitmask of the roles that a node fulfills.
-	pub struct Roles: u8 {
-		/// No network.
-		const NONE = 0b00000000;
-		/// Full node, does not participate in consensus.
-		const FULL = 0b00000001;
-		/// Light client node.
-		const LIGHT = 0b00000010;
-		/// Act as an authority
-		const AUTHORITY = 0b00000100;
+/// Role of the local node.
+#[derive(Debug, Clone)]
+pub enum Role {
+	/// Regular full node.
+	Full,
+	/// Regular light node.
+	Light,
+	/// Sentry node that guards an authority. Will be reported as "authority" on the wire protocol.
+	Sentry {
```

Comment:
I am not sure if I'd keep this one here since this enum should map to the "network role", and sentry isn't one, sentries should have role `Authority` and they should only exist as an `ObservedRole`. Pretty sure this variant actually never gets set when passing `--sentry`.

Reply:
Reading the rest of the PR maybe it makes sense to make the following distinction: `NetworkRole` (authority / full / light) > `LocalRole` (authority / full / light / sentry) > `ObservedRole` (authority / full / light / our_validator / our_sentry).

Reply:
> since this enum should map to the "network role",

I took the approach where the network role is now a concept internal to the network crate.
The `Role` enum in the network configuration now tells the network how it should behave, and the fact that we send our role to other nodes is purely an implementation detail.

Reply:
Okay I agree and I think it should work, just need to make sure we do set the sentry role.